snapcraft/commands/remote.py

```diff
@@ -243,3 +249,63 @@ def _get_effective_base(self) -> str:
             )
 
         return base
+
+    def _get_build_id(self) -> str:
```

Comment:
A question about the broader scope of the feature (remote build): eventually we want to move the related code somewhere other than snapcraft (possibly a new library, or craft-application) so that other tools can use it too, right? Will this build-id-related code also be moved? If so, do you think it would be worth it to refactor this a bit to have isolated functions (e.g. `get_build_id(app_name, project_name, project_path)`) that are easier to move later?

Reply:
I like that idea.  I can move this to `snapcraft/remote/utils.py` so it will be ready to migrate.